src/test/java/org/example/dao/MySqlJobRoleDaoTest.java

```diff
@@ -61,8 +74,53 @@ public void getJobRoles_shouldReturnListOfJobRoles_whenDatabaseReturnsRowsOfJobR
     @Test
     public void getJobRoles_shouldReturnListOfEmptyList_whenDatabaseReturnsNoRows()
             throws Exception {
+
         List<JobRole> jobRoles = IJobRoleDao.getJobRoles();
         assertNotNull(jobRoles);
         assertEquals(0, jobRoles.size());
     }
+
+    @Test
+    public void getJobRoleById_shouldReturnJobRoleDetails_whenDatabaseReturnsJobRole()
+            throws JobRoleDaoException, FileNotFoundException, SQLException,
+            ParseException {
+
+        try (Statement stmt = h2Connection.createStatement()) {
+            stmt.execute("DROP TABLE IF EXISTS roles");
+        }
+
+        RunScript.execute(h2Connection, new FileReader("src/test/resources/schema.sql"));
+        RunScript.execute(h2Connection, new FileReader("src/test/resources/data.sql"));
+
+        SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd");
+        Date closingDate = sdf.parse("2024-08-15");
+
+        JobRoleDetailsResponse jobRole = IJobRoleDao.getJobRoleById(1);
+        assertEquals("Software Engineer", jobRole.getRoleName());
+        assertEquals("London", jobRole.getLocation());
+        assertEquals("Full Stack Development", jobRole.getCapability());
+        assertEquals("Band 1", jobRole.getBand());
+        assertEquals(closingDate, jobRole.getClosingDate());
+        assertEquals(1, jobRole.getRoleStatus());
+        assertEquals("A Software Engineer develops", jobRole.getDescription());
+        assertEquals("Designing and developing software applications", jobRole.getResponsibilities());
+        assertEquals("http://example.com/job-link/SoftwareEngineer", jobRole.getJobLink());
+    }
+
+    @Test
+    public void getJobRoleById_shouldReturnEmptyJobRoleDetailsResponse_whenIdDoesNotExistInDatabase()
+            throws JobRoleDaoException {
+        JobRoleDetailsResponse emptyJobRoleDetailsObject = new JobRoleDetailsResponse();;
+        JobRoleDetailsResponse jobRole = IJobRoleDao.getJobRoleById(4);
+
+        assertEquals(emptyJobRoleDetailsObject.getRoleName(), jobRole.getRoleName());
+        assertEquals(emptyJobRoleDetailsObject.getLocation(), jobRole.getLocation());
+        assertEquals(emptyJobRoleDetailsObject.getCapability(), jobRole.getCapability());
+        assertEquals(emptyJobRoleDetailsObject.getBand(), jobRole.getBand());
+        assertEquals(emptyJobRoleDetailsObject.getClosingDate(), jobRole.getClosingDate());
+        assertEquals(emptyJobRoleDetailsObject.getRoleStatus(), jobRole.getRoleStatus());
+        assertEquals(emptyJobRoleDetailsObject.getDescription(), jobRole.getDescription());
+        assertEquals(emptyJobRoleDetailsObject.getResponsibilities(), jobRole.getResponsibilities());
+        assertEquals(emptyJobRoleDetailsObject.getJobLink(), jobRole.getJobLink());
```

Comment:
all of these assertions could just become this: `assertThat(jobRole).isEqualToComparingFieldByField(emptyJobRoleDetailsObject);`

Reply:
this was helpful lol thank you! Fixed